### PR TITLE
feat: Add `ClusterConfig`

### PR DESCRIPTION
**Pull request by dharanad**
Closes https://github.com/spacewalkhq/raft-rs/issues/26

* Added `ClusterConfig`, to provide Cluster infomation to server
* Removed SockAddr parsing from String rather added `SockAddr` to server config
* Lots of refactoring

### CODE REVIEW

File: src/cluster.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
+use std::collections::HashMap;
+use std::net::SocketAddr;
+
+#[derive(Debug, Clone)]
+pub struct NodeMeta {
```
Comment (dharanad) [inline src/cluster.rs:5]: My idea was each server will get it config from cluster config. Thought of moving `ServerConfig` here. However thats a big change (can be part of another PR). For now i have added `NodeMeta` is just a place holder.

File: src/server.rs [hunk 2]
```diff
@@ -902,28 +844,24 @@ impl Server {
...
 
         self.state.current_term = current_term;
         self.state.commit_index = commit_index;
         self.state.previous_log_index = previous_log_index;
 
-        self.peers.clear();
-        for i in 0..peers_count {
-            self.peers.push(i);
-        }
-
```
Comment (dharanad) [inline src/server.rs:915]: I believe we can find a more refined solution for this.

**Review by vaibhawvipul — APPROVED** [review]
